appservice/src/tree/AppSettingTreeItem.ts

```diff
@@ -23,15 +23,18 @@ export class AppSettingTreeItem extends AzureTreeItem<ISiteTreeRoot> {
     private _value: string;
     private _hideValue: boolean;
 
-    private constructor(parent: AppSettingsTreeItem, key: string, value: string) {
+    private readonly _client: IAppSettingsClient;
+
+    private constructor(parent: AppSettingsTreeItem, client: IAppSettingsClient, key: string, value: string) {
         super(parent);
+        this._client = client;
         this._key = key;
         this._value = value;
         this._hideValue = true;
     }
 
-    public static async createAppSettingTreeItem(parent: AppSettingsTreeItem, key: string, value: string): Promise<AppSettingTreeItem> {
-        const ti: AppSettingTreeItem = new AppSettingTreeItem(parent, key, value);
+    public static async createAppSettingTreeItem(parent: AppSettingsTreeItem, client: IAppSettingsClient, key: string, value: string): Promise<AppSettingTreeItem> {
```

Comment:
It's a bit late since it's merged, but why do you need to pass client as a parameter if you have it as a private property?